src/aiida/manage/configuration/profile.py

```diff
@@ -29,41 +31,44 @@
 class Profile:
     """Class that models a profile as it is stored in the configuration file of an AiiDA instance."""
 
-    KEY_UUID = 'PROFILE_UUID'
-    KEY_DEFAULT_USER_EMAIL = 'default_user_email'
-    KEY_STORAGE = 'storage'
-    KEY_PROCESS = 'process_control'
-    KEY_STORAGE_BACKEND = 'backend'
-    KEY_STORAGE_CONFIG = 'config'
-    KEY_PROCESS_BACKEND = 'backend'
-    KEY_PROCESS_CONFIG = 'config'
-    KEY_OPTIONS = 'options'
-    KEY_TEST_PROFILE = 'test_profile'
+    KEY_UUID: str = 'PROFILE_UUID'
+    KEY_DEFAULT_USER_EMAIL: str = 'default_user_email'
+    KEY_STORAGE: str = 'storage'
+    KEY_PROCESS: str = 'process_control'
+    KEY_STORAGE_BACKEND: str = 'backend'
+    KEY_STORAGE_CONFIG: str = 'config'
+    KEY_PROCESS_BACKEND: str = 'backend'
+    KEY_PROCESS_CONFIG: str = 'config'
+    KEY_OPTIONS: str = 'options'
+    KEY_TEST_PROFILE: str = 'test_profile'
 
     # keys that are expected to be in the parsed configuration
-    REQUIRED_KEYS = (
+    REQUIRED_KEYS: tuple[str, str] = (
         KEY_STORAGE,
         KEY_PROCESS,
     )
 
-    def __init__(self, name: str, config: Mapping[str, Any], validate=True):
+    def __init__(
+        self, name: str, config: abc.Mapping[str, Any], config_folder: pathlib.Path | None = None, validate: bool = True
```

Comment:
I am confused. You added a `config_folder` parameter here, but I don't see where you changed the places where the Profile objects are created?

Reply:
Yes, this can be removed. I put here just to show the profile class is independent of the default config folder. In principle, it should be possible to create a profile from the other config folder and mapping to the correct log/pid files.